src/Authentication.Abstractions/Interfaces/ICmdletContext.cs

```diff
@@ -0,0 +1,31 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+namespace Microsoft.Azure.Commands.Common.Authentication.Abstractions.Interfaces
+{
+    /// <summary>
+    /// Represents the context of a cmdlet.
+    /// </summary>
+    public interface ICmdletContext
+    {
+        /// <summary>
+        /// Gets the ID of the cmdlet.
+        /// </summary>
+        string CmdletId { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the cmdlet context is valid.
+        /// </summary>
+        bool IsValid { get; }
```

Comment:
Please help me understand why we need `IsValid`

Reply:
> Please help me understand why we need `IsValid`

It is to check whether the cmdletcontext is valid, in current implementation, whether it contains a valid cmdletId